bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/config/FtpFolderWatcherConfiguration.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.folderwatcher.internal.config;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * The {@link FtpFolderWatcherConfiguration} class contains fields mapping thing configuration parameters.
+ *
+ * @author Alexandr Salamatov - Initial contribution
+ */
+@NonNullByDefault
+public class FtpFolderWatcherConfiguration {
+    public String ftpAddress = "";
+    public int ftpPort;
```

Comment:
@J-N-K do I really need to do this? Isn't it redundant? What's the point of having default value in thing-types.xml then? I don't see this in other bindings.

Reply:
Which don‘t? Should be fixed there. The compiler can‘t know that and should show a warning if it is left uninitialized.

Reply:
https://github.com/openhab/openhab-addons/blob/2.5.x/bundles/org.openhab.binding.amazondashbutton/src/main/java/org/openhab/binding/amazondashbutton/internal/config/AmazonDashButtonConfig.java
or
https://github.com/openhab/openhab-addons/blob/2.5.x/bundles/org.openhab.binding.airquality/src/main/java/org/openhab/binding/airquality/internal/AirQualityConfiguration.java
I'm sure there are more

On the other hand some other bindings have initialization, so I'm confused. 
As I mentioned above even without specifying parameter value in thing config it is using default values as per thing-types.xml. Maybe openhab-core is handling this?

Anyway please let me know the final decision. I'll adjust accordingly.

Reply:
@J-N-K  did you have a chance to look at it?

Reply:
@J-N-K I'm waiting for a decision on this. Can we have another pair of eyes to review this? I don't think we need to initialize variables here since we have default values in thing-types.xml.